python/dgl/nn/mxnet/conv/graphconv.py

```diff
@@ -80,52 +93,75 @@ def __init__(self,
 
         self._activation = activation
 
-    def forward(self, graph, feat):
+    def forward(self, graph, feat, weight=None):
         r"""Compute graph convolution.
 
         Notes
         -----
-            * Input shape: :math:`(N, *, \text{in_feats})` where * means any number of additional
-              dimensions, :math:`N` is the number of nodes.
-            * Output shape: :math:`(N, *, \text{out_feats})` where all but the last dimension are
-              the same shape as the input.
+        * Input shape: :math:`(N, *, \text{in_feats})` where * means any number of additional
+          dimensions, :math:`N` is the number of nodes.
+        * Output shape: :math:`(N, *, \text{out_feats})` where all but the last dimension are
+          the same shape as the input.
+        * Weight shape: "math:`(\text{in_feats}, \text{out_feats})`.
 
         Parameters
         ----------
         graph : DGLGraph
             The graph.
         feat : mxnet.NDArray
             The input feature
+        weight : torch.Tensor, optional
+            Optional external weight tensor.
 
         Returns
         -------
         mxnet.NDArray
             The output feature
         """
         graph = graph.local_var()
-        if self._norm:
-            degs = graph.in_degrees().astype('float32')
-            norm = mx.nd.power(mx.nd.clip(degs, a_min=1, a_max=float("inf")), -0.5)
+
+        if self._norm == 'both':
+            degs = graph.out_degrees().as_in_context(feat.context).astype('float32')
+            degs = mx.nd.clip(degs, a_min=1, a_max=float("inf"))
+            norm = mx.nd.power(degs, -0.5)
             shp = norm.shape + (1,) * (feat.ndim - 1)
-            norm = norm.reshape(shp).as_in_context(feat.context)
+            norm = norm.reshape(shp)
             feat = feat * norm
 
+        if weight is not None:
+            if self.weight is not None:
+                raise DGLError('External weight is provided while at the same time the'
+                               ' module has defined its own weight parameter. Please'
+                               ' create the module with flag weight=False.')
+        else:
+            weight = self.weight.data(feat.context)
+
         if self._in_feats > self._out_feats:
             # mult W first to reduce the feature size for aggregation.
-            feat = mx.nd.dot(feat, self.weight.data(feat.context))
-            graph.ndata['h'] = feat
+            if weight is not None:
```

Comment:
weight would not be None here.
We have handled it in the above.

Reply:
It could be. We could create graphconv as `nn.GraphConv(..., weight=False)` where `self.weight` is None. When calling its forward function, does not provide external weight. This will result in computing a graph convolution without linear projection. Actually, this is used in the first layer of RGCN.